src/transport/SecureSessionMgr.cpp

```diff
@@ -266,6 +266,10 @@ CHIP_ERROR SecureSessionMgr::NewPairing(const Optional<Transport::PeerAddress> &
     {
         state->SetPeerAddress(peerAddr.Value());
     }
+    else if (peerAddr.HasValue() && peerAddr.Value().GetTransportType() == Transport::Type::kBle)
```

Comment:
It seems the logic we want is 'set peer address UNLESS it is ipaddress any. Checking IPAddress on transport type ble seems wrong, so maybe the correct condition should have been:

```cpp
if (peerAddr.HasValue()) {
   const auto& value = peerAddr.Value();
   if ((value.GetTransportType() != Transport::Type::kUdp && value.GetTransportType() != Transport::Type::kTcp) 
       || (value.GetIPAddress() != Inet::IPAddress::Any)) {
       state->SetPeerAddress(peerAddr.Value());
   }
}
```

some helper of peerAddr.Value().IsIpBased() may also make the code clearer. I would also reverse the order of conditions so that GetIPAddress() is never used for kBle.